Bir veriyi görselleştirebilmek için veriyi analiz etmeliyiz. Görselleştirme aslında analiz etme işlemidir.

Görselleştirme işlemi anlaşılabilirlik ve etkileyicilik açısından kullanılır.

https://www.kaggle.com/datasets/kwullum/fatal-police-shootings-in-the-us linkinde bir data mevcut. Bu data Amerika'da polisin insana ateş etmesi sonucunda ölümün gerçekleşmesi hakkındadır.

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

Veri setlerini çağıralım.

In [2]:
median_house_hold_in_come = pd.read_csv("MedianHouseholdIncome2015.csv", encoding="windows-1252")
percentage_people_below_poverty_level = pd.read_csv("PercentagePeopleBelowPovertyLevel.csv", encoding="windows-1252")
percent_over_25_completed_highSchool = pd.read_csv("PercentOver25CompletedHighSchool.csv", encoding="windows-1252")
share_race_city = pd.read_csv("ShareRaceByCity.csv", encoding="windows-1252")
kill = pd.read_csv("PoliceKillingsUS.csv", encoding="windows-1252")

Her bir eyaletin fakirlik oranı için percentage_people_below_poverty_level değişkeninde tutulan veriyi kullanalım.

In [3]:
percentage_people_below_poverty_level.head()

Geographic Area             City poverty_rate
0              AL       Abanda CDP         78.8
1              AL   Abbeville city         29.1
2              AL  Adamsville city         25.5
3              AL     Addison town         30.7
4              AL       Akron town           42

percentage_people_below_poverty_level değişkeninde tutulan veriyi inceleyelim. Geographic Area sütunu eyaleti ifade eder.
City sütunu şehri ve poverty_rate sütunu bu şehirdeki yoksulluk oranı bilgisini barındırır.

In [4]:
percentage_people_below_poverty_level.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29329 entries, 0 to 29328
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Geographic Area  29329 non-null  object
 1   City             29329 non-null  object
 2   poverty_rate     29329 non-null  object
dtypes: object(3)
memory usage: 687.5+ KB


In [5]:
percentage_people_below_poverty_level.poverty_rate.value_counts()

0       1464
-        201
7.4      129
6.7      129
10.9     128
        ... 
73.5       1
92         1
73.8       1
55.9       1
83.6       1
Name: poverty_rate, Length: 771, dtype: int64

Çıktıyı inceleyelim. poverty_rate yani fakirlik oranı 0 olan 1464 tane yer varmış. Yani 1464 yer fakir değil sonucuna varabiliriz.

İkinci satırda bulunan - 'nin anlamını bilmiyoruz. Bu durumda bu 201 tane veriyi görmezden gelmemiz gerekir. Bu - 'yi 0 'a eşitleyebiliriz.

In [6]:
percentage_people_below_poverty_level.poverty_rate.replace(['-'],0.0,inplace = True)

Yukarıdaki kod ile yapılan şudur: - değerini bulur ve - değerini 0 ile değiştirir. inplace = True parametresinden dolayı bu işlemi kalıcı bir şekilde kaydeder.

In [7]:
percentage_people_below_poverty_level.poverty_rate = percentage_people_below_poverty_level.poverty_rate.astype(float)

Yukarıdaki kod ile yapılan şudur: poverty_rate feature'sinin info metodu sonucunda gördüğümüz üzere object olduğunu anlıyoruz. Fakat bize number lazım. Dolayısıyla bu sütunu float'a çevirdik.

Böylece veri setinde amaçlarımıza uygun düzeltmeler gerçekleştirdik.